### ISSUE TITLE
Possible Blog Article: Importance of build/install to getting users and contributors. 

### ISSUE THREAD

**Issue by bernhold** [betterscientificsoftware/betterscientificsoftware.github.io#394]
I had an interesting lunchtime discussion at the SciDAC PI meeting.  One of the ideas that came up was the importance of the build/install experience to adoption of a software package.  In many cases, if it doesn't work, or is too hard to figure out, users will move on to something else, and they likely won't even bother providing feedback to the development team as to their problems.

Often, using a code is a gateway to contributing to a code, so a poor initial experience can turn way not only users, but also potential developers.

If we found someone from a project that has a really strong reputation for their attention to their build/install, perhaps they could write something along these lines.  I do not have any project or individual in mind at the moment.

**Comment by markcmiller86** [comment]
Great idea. I'd like to volunteer to help, minimally, as a reviewer.

I agree, the build and install experience is often the first experience would-be users have with a package. Some of the issues that come to mind are...
- Download
  - Does it require some kind of registration process?
  - Is the download so big it takes a long time?
  - Does it involve some non-standard compression tools?
- Is there an obvious place to "go" to find build instructions?
- Default build?
  - Is there one? Or, does building anything require specification of a slew of options
  - Are the default *reasonable* for common cases
  - Do common build options have non-standard usage (install prefix, compiler specification, etc.
  - Are non-standard tools needed for the build?
- Diagnosing build failures?
  - Is it even possible or are build logs hidden? Is there a clear place to go for help?
  - Is there an FAQ or list of commonly encountered issues?
- Is it easy to test an install and confirm its all working?

**Comment by rinkug** [comment]
@prwolfe and @bartlettroscoe to take an initial look and add comments

**Comment by bartlettroscoe** [comment]
This is an interesting topic.  A bad initial build experience will kill adoption in many cases.

**Comment by prwolfe** [comment]
I agree that this has been an issue for many (most) software I have been involved with in my career.   I would like to see this developed into a stronger proposal.

**Comment by rinkug** [comment]
There is interest on this topic.

**Comment by markcmiller86** [comment]
@asalmgren ... your recent experiences could be relevant here and I know we discussed doing something similar over lunch at ATPESC. Let us know with a comment here if you'd be interested in helping to craft a blog article on this topic.

**Comment by asalmgren** [comment]
Hi Mark -- yes, I'd be interested in working with you on this.

**Comment by rinkug** [comment]
David will be an EB member for this one
